hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java

```diff
@@ -683,15 +683,29 @@ public void closeContainer(long containerID) throws IOException {
   public Pipeline createReplicationPipeline(HddsProtos.ReplicationType type,
       HddsProtos.ReplicationFactor factor, HddsProtos.NodePool nodePool)
       throws IOException {
+    Map<String, String> auditMap = Maps.newHashMap();
+    if (type != null) {
+      auditMap.put("replicationType", type.toString());
+    }
+    if (factor != null) {
+      auditMap.put("replicationFactor", factor.toString());
+    }
+    if (nodePool != null) {
```

Comment:
```suggestion
    if (nodePool != null && !nodePool.getNodesList().isEmpty()) {
```
maybe also add an empty check?